lib/chef/provider/aws_iam_role_policy.rb

```diff
@@ -0,0 +1,36 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsIamRolePolicy < Chef::Provisioning::AWSDriver::AWSProvider
+  provides :aws_iam_role_policy
+
+  def action_create
+    iam_role_policy = super
+  end
+
+  protected
+
+  def create_aws_object
+    iam = new_resource.driver.iam_client
+
+    converge_by "create IAM role policy #{new_resource.name}" do
+      puts iam.put_role_policy({
+        role_name: new_resource.role,
+        policy_name: new_resource.name,
+        policy_document: new_resource.policy_document
+      }).inspect
+    end
+  end
+
+  def update_aws_object(iam_role_policy)
```

Comment:
It looks like you can update the inline policy document with http://docs.aws.amazon.com/sdkforruby/api/Aws/IAM/RolePolicy.html#put-instance_method